File: atv_academia/BancoDeDados.java
import java.util.ArrayList;
import java.util.HashSet;

public class BancoDeDados {
    private HashSet<Funcionario> funcionarios;

    public BancoDeDados(){
        this.funcionarios = new HashSet<>();
    }

    public void addFuncionarios(Funcionario funcionario) {
        this.funcionarios.add(funcionario);
    }

    public void removeFuncionarios(Funcionario funcionario) {
        this.funcionarios.remove(funcionario);
    }

    public void imprimirFuncionarios() {
        ArrayList<String> linhas = new ArrayList<>();
        for (Funcionario funcionario : this.funcionarios) {
            linhas.add(
                    String.format(
                            " Matrícula: %s Nome: %s Tipo: %s",
                            funcionario.getMatricula(),
                            funcionario.getNome(),
                            funcionario
                    )
            );
        }

        System.out.println("####### Funcionários #######################");
        for (String content : linhas) {
            System.out.println(content);
        }
        System.out.println("############################################");
    }

    public void imprimirProfessores() {
        ArrayList<Professor> professores = new ArrayList<>();
        for (Funcionario funcionario : this.funcionarios) {
            if (funcionario instanceof Professor){
                professores.add((Professor) funcionario);
            }
        }

        System.out.println("####### Professores ########################");
        for (Professor prof : professores) {
            System.out.println(String.format(" Professor [%s]", prof.getNome()));

            for (Aluno aluno : prof.getAlunos()) {
                System.out.println(String.format(" Aluno >>> %s %s", aluno.getNome(), aluno.getTreinosPrint()));
            }
        }
        System.out.println("############################################");
    }
}

File: atv_academia_kotlin/src/main/kotlin/BancoDeDados.kt
class BancoDeDados {
    private val funcionarios: HashSet<Funcionario>

    init {
        funcionarios = java.util.HashSet()
    }

    fun addFuncionarios(funcionario: Funcionario) {
        funcionarios.add(funcionario)
    }

    fun removeFuncionarios(funcionario: Funcionario) {
        funcionarios.remove(funcionario)
    }

    fun imprimirFuncionarios() {
        val linhas = ArrayList<String>()
        for (funcionario in funcionarios) {
            linhas.add(
                String.format(
                    " Matrícula: %s \t Nome: %s \t Tipo: %s",
                    funcionario.matricula,
                    funcionario.nome,
                    funcionario.tipoFuncionario
                )
            )
        }
        println("####### Funcionários #######################")
        for (content in linhas) {
            println(content)
        }
        println("############################################")
    }

    fun imprimirProfessores() {
        val professores = ArrayList<Professor>()
        for (funcionario in funcionarios) {
            if (funcionario is Professor) {
                professores.add(funcionario)
            }
        }
        println("####### Professores ########################")
        for (prof in professores) {
            println(String.format(" Professor [%s]", prof.nome))
            for (aluno in prof.getAlunos()) {
                println(String.format(" Aluno >>> %s %s", aluno.nome, aluno.treinosPrint))
            }
        }
        println("############################################")
    }

    fun imprimirRecepcionistas() {
        val recepcionista: ArrayList<Recepcionista> = ArrayList<Recepcionista>()
        for (funcionario in funcionarios) {
            if (funcionario is Recepcionista) {
                recepcionista.add(funcionario as Recepcionista)
            }
        }
        println("####### Recepcionista ########################")
        for (recep in recepcionista) {
            println(java.lang.String.format(" Nome : [%s] %s", recep.nome, recep.horariosPrint))
        }
        println("############################################")
    }
}
